### PR TITLE
[FIX] sms: autofocus on "body" if phone number is valid

### PR DESCRIPTION
**Pull request by nla-odoo**
Description of the issue/feature this PR addresses:
- As a user wanting to send SMS Text Message to a contact, I Edit the record and input a phone
 number Save Click on the SMS widget Expect to be able to write the message already

 - Instead, the focus is put on the phone number field again. This means that if I start typing,
 I'll actually "corrupt" the phone number, need to discard my change, and then click in the body
 to finally start typing my SMS Text Message.

Desired behavior after PR is merged:
- When the SMS composer opens, the focus will be put on the body field if the phone number is valid.

Task: https://www.odoo.com/web#id=2394488&action=4043&model=project.task&view_type=form&cids=2&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sms/wizard/sms_composer_views.xml [hunk 1]
```diff
@@ -48,8 +48,8 @@
                             <field name="recipient_single_number_itf" class="oe_inline" nolabel="1" options="{'onchange_on_keydown': True}" placeholder="e.g. +1 415 555 0100"/>
                         </div>
                                                 
-                        <field name="body" widget="sms_widget"/>
-                        <field name="mass_keep_log" invisible="1"/>
```
Comment (tde-banana-odoo) [inline addons/sms/wizard/sms_composer_views.xml:52]: Why did you remove it ? It allows to give default values even if invisible.

File: addons/sms/wizard/sms_composer_views.xml [hunk 2]
```diff
@@ -48,8 +48,8 @@
...
+                        <field name="body" widget="sms_widget" attrs="{'invisible': [('comment_single_recipient', '=', True), ('recipient_single_valid', '=', True)]}"/>
+                        <field name="body" widget="sms_widget" attrs="{'invisible': ['|', ('comment_single_recipient', '=', False), '&amp;', ('comment_single_recipient', '=', True), ('recipient_single_valid', '=', False)]}" default_focus="1"/>
```
Comment (tde-banana-odoo) [inline addons/sms/wizard/sms_composer_views.xml:52]: "not(A and B)" should be "not A or not B", or am I mistaken ?

**Review by tde-banana-odoo — COMMENTED** [review]
Validation

File: addons/sms/wizard/sms_composer_views.xml (hunk 2 above)
Comment (dja-odoo) [reply addons/sms/wizard/sms_composer_views.xml:52]: We have two condition for the OR part here. Not sure what you're trying to say.

Comment (tde-banana-odoo) [reply addons/sms/wizard/sms_composer_views.xml:52]: Second body is basically the inverse of first body. So I think domain of second one could be simplified.

**Comment by tde-banana-odoo** [comment]
@robodoo r+
